crates/oxc_isolated_declarations/src/diagnostics.rs

```diff
@@ -139,3 +139,8 @@ pub fn type_containing_private_name(name: &str, span: Span) -> OxcDiagnostic {
     ))
     .with_label(span)
 }
+
+pub fn destructuring_export(span: Span) -> OxcDiagnostic {
+    OxcDiagnostic::error("TS9999: Not supported yet")
```

Comment:
I wanted to give feedback, that a certain transformation is not supported yet, instead of silently generating wrong `.d.ts`

Reply:
I agree, can you provide the examples that need to report this error?

Reply:
The `TSXXXX` error code is identical to the error reported by typescript. If you want to report an error that typescript doesn't have, you don't need to add `TSXXXX`.

Reply:
> can you provide the examples that need to report this error?

1. I added it [here](https://github.com/escaton/oxc/blob/bc09ef80aa5830f65bff90396bb3571efa2de25f/crates/oxc_isolated_declarations/src/declaration.rs#L59) because `return None;` leads to `export const {a} = b()` is transformed to `export declare const;` instead of `export declare const a: unknown;`.
2. I also considered adding it [here](https://github.com/escaton/oxc/blob/8fd3ae99af82d1bc107f9e7a3ee42204b2365df9/crates/oxc_isolated_declarations/src/class.rs#L344) because with current `// Skip implementation of function overloads` it generates incorrect code, and I don't have ideas how to fix it right away.

Reply:
> > can you provide the examples that need to report this error?
> 
> 1. I added it [here](https://github.com/escaton/oxc/blob/bc09ef80aa5830f65bff90396bb3571efa2de25f/crates/oxc_isolated_declarations/src/declaration.rs#L59) because `return None;` causes `export const {a} = b()` to be transformed into `export declare const;` instead of `export declare const a: unknown;`.

This is a bug. Our goal is to keep the output the same as TypeScript, including diagnostics. if it's not the same, it's considered a bug. So we should report `binding_element_export` here

> 2. I also considered adding it [here](https://github.com/escaton/oxc/blob/8fd3ae99af82d1bc107f9e7a3ee42204b2365df9/crates/oxc_isolated_declarations/src/class.rs#L344) because with current `// Skip implementation of function overloads` it generates incorrect code, and I don't have ideas how to fix it right away.
> 
> Methods in abstract classes could be empty, and it is recognized as overloading:
> 
> ```
> export abstract class A {
>   b(): void;
>   c(): void {};
>   d(): void {}
> }
> ```
> 
> ->
> 
> ```
> export abstract class A {
>   b(): void;
>   d(): void {}
> }
> ```

This code seems to be incorrect. [TypeScript Playground](https://www.typescriptlang.org/play/?isolatedDeclarations=true#code/KYDwDg9gTgLgBAQwEYGcZQQY3pgNglFOAQTgG8AoOapACgEoAuOANwgEsATAbiuswbM2XcgF9e1OJ0GsOnMRVEUgA). We assume that the transforms code is correct

Reply:
The first case is fixed in #3976, The second case has a similar [issue](https://github.com/oxc-project/oxc/pull/4019), and fixed in #4024

Reply:
fixed in https://github.com/oxc-project/oxc/pull/4025, right?

Reply:
Yes!!